### PR TITLE
Add labels and annotations to attestation

### LINKED ISSUE #591: Provide TaskRun annotations in SLSA provenance
### Feature request

I'm mostly interested in having the `experimental.tekton.dev/execution-mode` annotation available in the SLSA provenance. It might be useful to place all TaskRun annotations in there as well. A good place might be `invocation.environment.context`, similar to how this is done for GitHub Workflows in the [this example](https://slsa.dev/provenance/v0.2#github-actions).

### Use case

I want to see from the SLSA provenance if a build was done in hermetic environment or not. This allows me to evaluate the built result with higher level of confidence that there wasn't outside tampering with certain TaskRun. For example, for the build, having fetched the source code and having all the dependencies vendored I don't want to allow any external traffic, and I'd like to attest to that in the SLSA provenance generated for that TaskRun.

This was already proposed in #278, though it was closed as stale.

### PR DESCRIPTION
**Pull request by lcarva**
Fixes #591

# Changes

Labels and annotations are added to the PipelineRun and TaskRun attestations in `.predicate.invocation.environment`. Additionally, for PipelineRun attestations, each task also contains its own `.invocation.environment` with the corresponding labels and annotations.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
- Annotations and labels are included in the attestation.
```

### CODE REVIEW

File: test/testdata/intoto/pipeline-output-image.json [hunk 1]
```diff
@@ -42,7 +45,12 @@
                     ],
                     "invocation": {
                         "configSource": {},
-                        "parameters": {}
+                        "parameters": {},
+                        "environment": {
+                            "annotations": {
+                                "pipeline.tekton.dev/release": "9221ac3-dirty"
```
Comment (chitrangpatel) [inline test/testdata/intoto/pipeline-output-image.json:51]: Is this value hard-coded `9221ac3-dirty`? Just checking. Probably don't want this to change if we update the installation of Tekton pipelines in CI.

**Review by chitrangpatel — APPROVED** [review]
Just one nit to double check.

File: test/testdata/intoto/pipeline-output-image.json (hunk 1 above)
Comment (lcarva) [reply test/testdata/intoto/pipeline-output-image.json:51]: You're right... I think I'll have to delete this attribute from the annotations before the call to Diff.

Comment (wlynch) [reply test/testdata/intoto/pipeline-output-image.json:51]: You can have cmp.Diff ignore fields with this! https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#IgnoreFields

File: pkg/chains/formats/slsa/attest/attest.go [hunk 2]
```diff
@@ -74,6 +76,30 @@ func Invocation(source *v1beta1.ConfigSource, params []v1beta1.Param, paramSpecs
 	}
 
 	i.Parameters = iParams
+
+	environment := map[string]map[string]string{}
+
+	annotations := map[string]string{}
+	for name, value := range meta.GetAnnotations() {
+		// Ignore annotations that are not relevant to provenance information
+		if name == corev1.LastAppliedConfigAnnotation || strings.HasPrefix(name, "chains.tekton.dev/") {
```
Comment (wlynch) [inline pkg/chains/formats/slsa/attest/attest.go:85]: Thanks for thinking of this! This seems reasonable, but we should probably document this behavior somewhere so it doesn't catch someone by surprise.

**Review by wlynch — COMMENTED** [review]
Mostly LGTM! Just some minor comments.

File: pkg/chains/formats/slsa/attest/attest.go (hunk 2 above)
Comment (lcarva) [reply pkg/chains/formats/slsa/attest/attest.go:85]: Good idea. Added

File: test/testdata/intoto/pipeline-output-image.json (hunk 1 above)
Comment (lcarva) [reply test/testdata/intoto/pipeline-output-image.json]: Done

**Review by wlynch — APPROVED** [review]
/lgtm

**Pull request merged**